samples/cpp/hello_world/src/main.cpp

```diff
@@ -0,0 +1,13 @@
+/*
+ * Copyright (c) 2023, Meta
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <iostream>
+
+int main(void)
+{
+	std::cout << "Hello, C++ world! " << CONFIG_BOARD << std::endl;
```

Comment:
not a request for change but a comment: I know this was discussed before so if you have evidence that people should avoid `std::cout` in real-life, might be worth mentioning? FWIW it does not seem to really bloat the ROM and RAM footprint at all compared to a simple printk, so there's that I guess.

Reply:
Mentioned before, but there is plenty of literature out there that says to avoid using `iostream` in an embedded environment. I don't need to repeat it here.

Also, that would preclude the use case where something like e.g. pigweed is used as a size-optimized replacement (assuming that can be done relatively easily by include path).

I think it's nice too, to simply support the de-facto C++ hello world (if we can) because it lowers the barrier to entry by using the standard approach, which is mostly the goal of hello world.